.github/workflows/ci.yml

```diff
@@ -14,35 +14,60 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 jobs:
-  build:    
+  prepare:
+    runs-on: macos-12
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/build
+        with:
+          project-id: ${{ secrets.PROJECT_ID }}
+
+  test:    
+    needs: prepare
     runs-on: macos-12
     strategy:
+      fail-fast: false
       matrix:
-        test-type: [unit-tests, integration-tests, build-example-wallet, build-example-dapp, relay-tests]
+        test-type: [integration-tests, smoke-tests, relay-tests]
```

Comment:
smoke tests does not have to run together with integration tests.
smoke-tests is just a test plan with selected integration tests
we want to run all integration tests on our CI and smoke tests in rust workflow


why build-example-wallet, build-example-dapp are excluded now?
we need to build apps to check if sdk did not break integration

Reply:
I think we should run different workflows on pr to main and pr to develop

Reply:
btw, rust is running against our release, so smoke tests can be build for testing on pr to main only

Reply:
Oh right, We don't need to run both smoke_tests and integration_tests as smoke_tests are a subset.

Reply:
for the build-example-wallet and build-example-dapp, I think this can now be verified in the prepare step when calling `build-for-testing`. I will double-check what's more efficient, having it built as part of that or having individual jobs like it was before.

Reply:
@llbartekll can you elaborate on this? Do you mean running only specific tests when targeting develop and different tests when targeting main? 

> I think we should run different workflows on pr to main and pr to develop

Reply:
I mean that Rust relay needs to verify it's staging works with our prod before they can deploy.
and they run our smoke tests, so the build for testing(smoke_tests) artifact needs to reflect our prod, which is main branch.
So we don't need to build for testing on develop merges as it does not reflect our prod.